### PR TITLE
Remove dependency on hyper 0.10 and a few other old crates

### PR DESCRIPTION
**Pull request by mzohreva**
- Remove dependency on `hyper` 0.10 and use `simple-hyper-client` instead
- Remove dependency on `rustc-serialize` and use `base64` instead
- Remove dependency on `chrono` and use `time` instead
- Run `cargo audit` in CI
- Bump up version from 0.3.0 to 0.4.0 due to breaking changes in API (i.e. `Time` and `SdkmsClientBuilder`)

### CODE REVIEW

**Comment by mzohreva** [comment]
bors try

File: src/client.rs [hunk 1]
```diff
@@ -148,16 +140,16 @@ impl SdkmsClientBuilder {
 /// which point you can call [`request_approval_to_encrypt()`]. There is an example of how to do this in
 /// [the repository](https://github.com/fortanix/sdkms-client-rust/blob/master/examples/approval_request.rs).
 ///
-/// [`hyper::Client`]: https://docs.rs/hyper/0.10/hyper/client/struct.Client.html
-/// [`hyper_native_tls::NativeTlsClient`]: https://docs.rs/hyper-native-tls/0.3/hyper_native_tls/struct.NativeTlsClient.html
+/// [`simple_hyper_client::blocking::Client`]: TODO
```
Comment (mzohreva) [inline src/client.rs]: Fix TODO

**Comment by mzohreva** [comment]
bors try

**Review by Pagten — APPROVED** [review]

**Review by MihirLuthra — APPROVED** [review]
Looks good to me.

**Pull request merged**